frontend/src/components/InputText.js

```diff
@@ -1,8 +1,76 @@
 import React, {Component} from 'react';
+import InputTextItems from './InputTextItems'
+import './InputText.css'
 
 class InputText extends Component {
+    constructor(properties) {
+        super(properties);
+       
+        this.state = {
+          items: []
+        };
+       
+        this.addItem = this.addItem.bind(this);
+        this.deleteItem = this.deleteItem.bind(this);
+
+    }
+
+    addItem(event) {
+      if (this._inputElement.value === "") {
+        return;
+      }
+      const name = this._inputElement.value;
+      const nameCapitalized = name.charAt(0).toUpperCase() + name.slice(1).toLowerCase();
+      var isDuplicate = false;
+      var newItem = {
+        text: nameCapitalized,
+        key: Date.now()
+      };
+      this.setState((prevState) => {
+        prevState.items.forEach(function(item, index, object) {
+          if (item.text===newItem.text){
+            isDuplicate=true;
+            window.alert("Duplicate Ingredient");
+          }
+        });
+        if (!isDuplicate) {
```

Comment:
nit: positive conditionals are easier to parse. Flip this block like so:
if (isDuplicate) {

} else {

}

Or better, use a ternary statement:
(untested code)
return {
 items: isDuplicate ? items: prevState.items : [newItem].concat(prevState.items)
};